game_patch/multi/server.cpp

```diff
@@ -657,6 +665,26 @@ CodeInjection multi_limbo_init_injection{
     },
 };
 
+CallHook<void(const char* filename)> multi_change_level_call_hook{
+    0x0046E89C,
```

Comment:
`multi_change_level` is called from 3 offsets in `multi_check_for_round_end` function. If you open assembly you will see. Therefore this `CallHook` is not going to be executed in all situations.
Also I don't like the place where you installed hook. The game has all level rotation related logic in `multi_load_next_level` function (`++netgame.current_level_index;`) and I would expect hook for that function or injection in that function. 
I think it would be best to use `CodeInjection` and if random rotation is enabled replace `++netgame.current_level_index;` with something like `netgame.current_level_index = rand() % num_levels`

Reply:
> I think it would be best to use CodeInjection and if random rotation is enabled replace ++netgame.current_level_index; with something like netgame.current_level_index = rand() % num_levels

This was originally the way I was approaching it, but it had the side effect of also overriding issuing of map_next and such, so I tried to go directly to the location where the level that is to be loaded is set. It's definitely possible I'm confusing my logic :P but that was why I didn't do it that way.